api/v1/cluster_types.go

```diff
@@ -2333,20 +2348,30 @@ type ExternalCluster struct {
 	BarmanObjectStore *BarmanObjectStoreConfiguration `json:"barmanObjectStore,omitempty"`
 }
 
-// AppendAdditionalCommandArgs adds custom arguments as barman cloud command-line options
-func (cfg *BarmanObjectStoreConfiguration) AppendAdditionalCommandArgs(options []string) []string {
-	if cfg == nil || cfg.Data == nil {
+// AppendAdditionalCommandArgs adds custom arguments as barman-cloud-backup command-line options
+func (cfg *DataBackupConfiguration) AppendAdditionalCommandArgs(options []string) []string {
+	if cfg == nil {
 		return options
 	}
+	return appendAdditionalCommandArgs(cfg.AdditionalCommandArgs, options)
+}
 
-	for _, userOption := range cfg.Data.AdditionalCommandArgs {
+// AppendAdditionalCommandArgs adds custom arguments as barman-cloud-wal-archive command-line options
+func (cfg *WalBackupConfiguration) AppendAdditionalCommandArgs(options []string) []string {
```

Comment:
@pselliotc PR looks good to me.
Do you mind if I ask you to add some unit test cases for this method `AppendAdditionalCommandArgs` inside `cluster_types_test.go`?
Same for `appendAdditionalCommandArgs`.

Reply:
Yes of course! I've added unit tests for both `AppendAdditionalCommandArgs` methods and the private `appendAdditionalCommandArgs` method.